vunit/source_file.py

```diff
@@ -8,11 +8,10 @@
 Functionality to represent and operate on VHDL and Verilog source files
 """
 from pathlib import Path
-from typing import Union
+from typing import Union, Dict
 import logging
 from copy import copy
 import traceback
-from vunit.sim_if.factory import SIMULATOR_FACTORY
```

Comment:
Why was this imported to the funcs/methods? It has the potential to be imported many, many times now which seems pointless to me.

Reply:
We were having an issue with it for some reason. I have it on my list of things to clean up/fix.